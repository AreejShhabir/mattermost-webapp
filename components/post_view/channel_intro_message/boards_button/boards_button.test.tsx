// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import BoardsButton from './boards_button';

const component = (otherProps: any) => (
    <BoardsButton
        boardComponent={
            <button>{'create a board'}</button>
        }
        show={true}
        {...otherProps}
    />
);

describe('components/post_view/ChannelIntroMessages', () => {
    describe('test Boards Button', () => {
        test('should match snapshot', () => {
            expect(shallow(component({}))).toMatchSnapshot();
        });

        test('should match snapshot, no boards', () => {
            expect(shallow(component({
                boardComponent: null,
            }))).toMatchSnapshot();
        });

        test('should match snapshot, has board, no show', () => {
            expect(shallow(component({
                show: false,
            }))).toMatchSnapshot();
        });
    });
});
