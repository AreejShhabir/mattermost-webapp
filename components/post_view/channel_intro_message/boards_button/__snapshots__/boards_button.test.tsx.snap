// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/ChannelIntroMessages test Boards Button should match snapshot 1`] = `
<button
  aria-label="Create a board"
  className="intro-links color--link channelIntroButton style--none"
  onClick={[Function]}
>
  Create a board
</button>
`;

exports[`components/post_view/ChannelIntroMessages test Boards Button should match snapshot, no boards 1`] = `""`;
